pkg/apis/pipeline/v1beta1/image_source.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright 2018 The Knative Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+// ImageSource defines an endpoint where artifacts can be stored, such as images.
+type ImageSource struct {
+	Name string `json:"name"`
+	// TODO: maybe an enum, with values like 'registry', GCS bucket
+	Type string `json:"type"`
+	URL  string `json:"url"`
```

Comment:
I think it would maybe be useful to separate the endpoint where the image is tored from the URL - we might want to have an endpoint, tag, and sha all available somehow

Reply:
Thats would be the metadata for image source. We could consider `metadata` for source's interface to provide this information.

Reply:
How about we use the subset of fields Redhat Atomic Host Signature. https://github.com/aweiteka/image/blob/e5a20d98fe698732df2b142846d007b45873627f/docs/signature.md 
```
 "identity": {
            "docker-reference": "busybox"
        },
        "image": {
            "docker-manifest-digest": "sha256:a59906e33509d14c036c8678d687bd4eec81ed7c4b8ce907b888c607f6a1e0e6"
        }, 
```
Some Missing fields are "Registry"

Reply:
`docker-reference` could be combination of `registry:docker-image-name` as well.